server/models/meshmodel/registration/register.go

```diff
@@ -0,0 +1,157 @@
+	Errors are written to the log file.
+*/
+func (rh *RegistrationHelper) Register(entity RegisterableEntity) {
+	// get the packaging units
+	pu, err := entity.PkgUnit()
+	if(err != nil){
+		// given input is not a valid model, or could not walk the directory
+		return
+	}
+	// fmt.Printf("Packaging Unit: Model name: %s, comps: %d, rels: %d\n", pu.model.Name, len(pu.components), len(pu.relationships))
+	rh.register(pu)
+}
+
+
+/*
+	register will return an error if it is not able to register the `model`.
+	If there are errors when registering other entities, they are handled properly but does not stop the registration process.
+*/
+func (rh *RegistrationHelper)register(pkg packagingUnit) {
+	// 1. Register the model
+	model := pkg.model
+	if(model.Status != entity.Enabled){return}
+
+	// Dont register anything else if registrant is not there
+	if(model.Registrant.Hostname == ""){
+		err := ErrMissingRegistrant()
+		RegLog.InsertEntityRegFailure(model.Registrant.Hostname, "",entity.Model, model.Name, err)
+		return
+	}
+	_, _, err := rh.regManager.RegisterEntity(
+		v1beta1.Host{Hostname: model.Registrant.Hostname,},
+		&model,
+		)
+
+	// If model cannot be registered, don't register anything else
+	if err != nil {
+		err = ErrRegisterEntity(err, string(model.Type()), model.DisplayName)
+		RegLog.InsertEntityRegFailure(model.Registrant.Hostname, "",entity.Model, model.Name, err)
+		return
+	}
+
+	hostname := model.Registrant.Hostname
+	modelName := model.Name
+	// 2. Register components
+	for _, comp := range pkg.components {
+		comp.Model = model
+		if(comp.Model.Status != entity.Enabled){
+			continue
+		}
+		_, _, err := rh.regManager.RegisterEntity(
+		v1beta1.Host{Hostname: hostname,},
+		&comp,
+		)
+	if err != nil {
+		err = ErrRegisterEntity(err, string(comp.Type()), comp.DisplayName)
+		RegLog.InsertEntityRegFailure(hostname, modelName ,entity.ComponentDefinition, comp.DisplayName, err)
+		rh.log.Error(err)
+	}
+	}
+
+	// 3. Register relationships
+	for _, rel := range pkg.relationships {
+		rel.Model = model
+		_, _, err := rh.regManager.RegisterEntity(v1beta1.Host{
+			Hostname: hostname,
+		}, &rel)
+		if err != nil {
+			err = ErrRegisterEntity(err, string(rel.Type()), rel.Kind)
+			rh.log.Error(err)
+			RegLog.InsertEntityRegFailure(hostname, modelName ,entity.RelationshipDefinition, rel.ID.String(), err)
+		}
+	}
+}
+
+func (erh *RegistrationHelper) RegistryLog() {
+	log := erh.log
+	provider := erh.handlerConfig.Providers["None"]
+
+	systemID := viper.GetString("INSTANCE_ID")
+
+	sysID := uuid.FromStringOrNil(systemID)
+	hosts, _, err := erh.regManager.GetRegistrants(&v1beta1.HostFilter{})
+	if err != nil {
+		log.Error(err)
+	}
+
+	for _, host := range hosts {
+		eventBuilder := events.NewEvent().FromSystem(sysID).FromUser(sysID).WithCategory("entity").WithAction("get_summary")
+		successMessage := fmt.Sprintf("For registrant %s successfully imported", host.Hostname)
```

Comment:
```suggestion
		successMessage := fmt.Sprintf("For registrant %s imported", host.Hostname)
```

How many imported? -- -- see the predefined error message in MeshKit.

Reply:
Updated it